data_manipulation/dates_times.Rmd

```diff
@@ -0,0 +1,335 @@
+This section shows how to use `lubridate` together with `dplyr`.
+
+The dataset `temperature` includes a few days of air temperature from the [GFI weather station](https://veret.gfi.uib.no/), Bergen.
+
+The dataset has three columns, `Dato`, `Tid` and `Lufttemperatur`. 
+I have imported the data with `readr::read_csv` so the first two columns have automatically been converted to date and time formats. 
+You can tell this because the class of each column is given below the column name.
+
+```{r print-temperature}
+temperature
+```
+
+### Using `mutate` to make a new column
+
+I want to make a single date-time column. 
+To do this I need to combine the data and time columns with `paste`, and then use a `lubridate` function for the conversion. 
+
+```{r date-time-temperature}
+temperature2 <- temperature %>% 
+  mutate(
+    date_time = paste(Dato, Tid), #combine
+    date_time = ymd_hms(date_time)#convert
+    ) 
+
+temperature2
+```
+
+`lubridate` functions `make_date` and `make_datetime` are useful if the year, month, day etc. are in different columns.
+
+```{r make-date}
+tibble(year = 2020, month = 10, day = 3) %>% 
+  mutate(date = make_date(year = year, month = month, day = day))
+```
+
+
+### Using `filter` to select rows
+
+I often need to `filter` data to remove bad data or restrict the dates and times in the dataset.
+
+For example, to calculate the mean temperature of the afternoon of the 21st April, we first need to `filter` the data. 
+`between` is a useful helper function here to avoid having to write a more complicated logical condition.
+
+```{r filter-temperature}
+temperature2 %>% 
+  filter(between(date_time, 
+                 left = ymd_hm("2020-04-21 12:00"),
+                 right = ymd_hm("2020-04-21 17:00"))) %>% 
+  summarise(mean_temp = mean(Lufttemperatur))
+
+```
+
+
+### Using `group_by` and `summarise`
+
+I often want to summarise climate data to calculate mean monthly temperatures. 
+We can do an analogous task with the `temperature` dataset to calculate mean hourly temperatures.
+
+```{r mean-hourly}
+temperature2 %>% 
+  mutate(hour = hour(date_time)) %>% 
+  group_by(hour) %>% 
+  summarise(mean_temperature = mean(Lufttemperatur))
+```
+
+
+## Plotting dates
+
+We can plot the temperature data with `ggplot`.
+
+```{r}
+ggplot(temperature2, aes(x = date_time, y = Lufttemperatur)) + 
+  geom_line()
+```
+
+`ggplot` recognises that the x-axis variable is a date-time and plots the axis with appropriate labels. 
+The language will depend on the locale (see `Sys.getlocale()`). 
+I don't like the way the dates have been formatted so I can use `scale_x_datetime` to alter the format. 
+The codes from `?strptime` that you learnt at the start of this tutorial finally become useful!
+
+```{r}
+ggplot(temperature2, aes(x = date_time, y = Lufttemperatur)) + 
+  geom_line() +
+  scale_x_datetime(date_labels = "%A %d %B") #see also date_breaks to change label positions
+```
+
+
+We can also plot times. 
+
```

Comment:
not sure what you mean with times? Everything is about times here

Reply:
bare times without dates. 15:34. with hms class. Clarified